core/htmlparser/filter.js

```diff
@@ -402,6 +402,36 @@
 } )();
 
 /**
+ * This is an abstract class representing filter rules passed to {@link CKEDITOR.htmlParser.filter#constructor} as an argument. See {@link CKEDITOR.htmlParser.filter} for examples.
```

Comment:
I'm not sure if limiting this class to filter constructor is a good idea. This class may be used in different places - moreover, it's already used as an argument for [addRules](https://ckeditor.com/docs/ckeditor4/latest/api/CKEDITOR_htmlParser_filter.html#method-addRules) function. Also in the future, we may want to use it in other places which with the proposed documentation shape will require another refactoring.

Also, mentioning that this class is abstract and used in some other places is redundant. This information is already viewable from functions arguments and returned types.

Instead of linking `htmlParser.filter` for examples, I think that it would be nice to attach some example code of definition and refactor duplicated code from filter listing.

Reply:
I would extend this description with the information like:

What this class represents?
Why would I have to use it?
How I have to use it?

not implementation details.